common/scripts/x86_64/RHEL_7/_helpers.sh

```diff
@@ -340,6 +344,58 @@ __pull_images() {
   done
 }
 
+__pull_admiral_image() {
+  __process_marker "Pulling latest admiral image"
+  __process_msg "Registry: $PRIVATE_IMAGE_REGISTRY"
+  __registry_login
+
+  image="$PRIVATE_IMAGE_REGISTRY/$ADMIRAL_IMAGE:$RELEASE"
+  sudo docker pull $image
+}
+
+__pull_images_master() {
+  __process_marker "Pulling latest service images on master"
+  __process_msg "Registry: $PRIVATE_IMAGE_REGISTRY"
+  __registry_login
+
+  local master_ip=""
+
+  if [ "$DEV_MODE" != "true" ]; then
+    local get_master_query="PGPASSWORD=$DB_PASSWORD \
+      psql \
+      -U $DB_USER \
+      -d $DB_NAME \
+      -h $DB_IP \
+      -p $DB_PORT \
+      -v ON_ERROR_STOP=1 \
+      -tc 'SELECT master from \"systemSettings\"; '"
+
+    {
+      local master=`eval $get_master_query` &&
+      __process_msg "'systemSettings' table exists, finding master IP"
+      master_ip=$(echo "$master" | jq '.address')
+    } || {
+      __process_msg "'systemSettings' table does not exist, skipping"
+    }
+  fi
+
+  for image in "${SERVICE_IMAGES[@]}"; do
+    image="$PRIVATE_IMAGE_REGISTRY/$image:$RELEASE"
+    local pull_cmd="sudo docker pull $image"
+    if [ -z "$master_ip" ]; then
+      __process_msg "Pulling $image"
+      sudo docker pull $image
+    else
+      # TODO: have a function which returns the docker login command instead of figuring it out everytime
+      local docker_login_cmd="aws ecr --no-include-email --region us-east-1 get-login | bash"
```

Comment:
This needs to also support the `--no-ssl-verify` flag. Something similar to https://github.com/Shippable/admiral/blob/537c1549802c9876e4d3906f803213e37cca5672/common/scripts/lib/_helpers.sh#L246.